distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Manager.java

```diff
@@ -86,6 +99,49 @@ public void start() throws Exception {
 
         leaderLatch.start();
         workersAliveChildrenCache.start();
+        startWorkPoolCleaningTask();
+    }
+
+    private void startWorkPoolCleaningTask() {
+            workPoolCleaningReschedulableScheduler.schedule(
+                    DynamicProperty.delegated(() -> Schedule.withDelay(workPoolCleanPeriodMs.get())),
+                    () -> {
+                        try {
+                            TransactionalClient.tryCommit(
+                                    curatorFramework,
+                                    CLEAN_WORK_POOL_RETRIES_COUNT,
+                                    transaction -> {
+
+                                        String workPoolVersion = paths.availableWorkPoolVersion();
+                                        int version = curatorFramework.checkExists().forPath(workPoolVersion).getVersion();
+                                        transaction.checkPathWithVersion(workPoolVersion, version);
+                                        transaction.setData(workPoolVersion, new byte[]{});
+
+                                        cleanWorkPool(transaction);
+                                    });
+                        } catch (Exception e) {
+                            log.debug("Failed to clean work-pool", e);
+                        }
+                    }
+            );
+    }
+
+    private void cleanWorkPool(TransactionalClient transaction) throws Exception {
+        if(log.isTraceEnabled()){
+            log.trace("cleanWorkPool zk tree before cleaning: {}", buildZkTreeDump());
+        }
+
+        Set<String> actualJobs = new HashSet<>();
+        for(String workerId: getChildren(paths.aliveWorkers())){
```

Comment:
why you are using getChildren(aliveWorkers) if you already have workersAliveChildrenCache,   
use workersAliveChildrenCache to access children, this way we will not  pollute zk by redundant requests

Reply:
ZK call replaced with using workersAliveChildrenCache. Maybe we should just return from work-pool cleaning method, instead of blocking the thread? It makes code a little bit easier